src/test/compile-fail/extern-types-unsized.rs

```diff
@@ -30,14 +30,14 @@ fn assert_sized<T>() { }
 
 fn main() {
     assert_sized::<A>();
-    //~^ ERROR the trait bound `A: std::marker::Sized` is not satisfied
+    //~^ ERROR `A` does not have a constant size known at compile-time
```

Comment:
something about this phrasing feels repetitive (if the size is known at compile-time, it must be constant?) or maybe just odd -- it's really that *values* of type `A` do not have a known size -- maybe just drop "constant"?

- `A` does not have a size known at compile-time

I also tried to rephrase in various ways, but I wasn't happy with them:

- the compiler cannot compute the size for values of type `A`
- values of type `A` do not have constant size
- the size for value values of type `A` cannot be known at compilation time

Reply:
Changed wording:

```
error[E0277]: the size for value values of type `str` cannot be known at compilation time
  --> $DIR/const-unsized.rs:22:26
   |
LL | static STATIC_BAR: str = *"bar";
   |                          ^^^^^^ doesn't have a size known at compile-time
   |
   = help: the trait `std::marker::Sized` is not implemented for `str`
   = note: to learn more, visit <https://doc.rust-lang.org/book/second-edition/ch19-04-advanced-types.html#dynamically-sized-types--sized>
   = note: constant expressions must have a statically known size
```